sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnectionStringBuilder.cs

```diff
@@ -29,11 +29,12 @@ public class ServiceBusConnectionStringBuilder
         const string OperationTimeoutConfigName = "OperationTimeout";
 
         string entityPath, sasKeyName, sasKey, sasToken, endpoint;
-        string authType;
+        AuthenticationType authType = AuthenticationType.None;
 
-        public static class AuthenticationType
+        public enum AuthenticationType
         {
-            public const string ManagedIdentity = "Managed Identity";
+            None,
```

Comment:
>None [](start = 12, length = 4)

This is interesting. For customers who are using SASTokens or SASKeys, this authenticationType would be None. Which is quite confusing.

Reply:
What is EH library doing here?

---
In reply to: [297837863](https://github.com/Azure/azure-sdk-for-net/pull/6637#discussion_r297837863) [](ancestors = 297837863)

Reply:
The initial implementation is also string-based, but it is in the process of being given an overhaul, where the new API surface will simply take a token credential that the `Azure.Identity` library is responsible for producing.    As a general rule, authentication should be an external concern;  authorization against the resource is an internal one.